libcaliptra/test/hwmod_test/hwmod_if_impl.c

```diff
@@ -0,0 +1,82 @@
+//Licensed under the Apache-2.0 license
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#include "caliptra_model.h"
+#include "caliptra_api.h"
+
+#define CALIPTRA_STATUS_OK 0
+
+// Implementation specifics
+
+struct caliptra_model_init_params init_params;
+struct caliptra_fuses fuses = {0};
+
+static bool caliptra_model_init_complete = false;
+static struct caliptra_model *model = NULL;
+
+struct caliptra_model* hwmod_get_or_init(void)
+{
+    if (model == NULL)
+    {
+        int status = caliptra_model_init_default(init_params, &model);
+    }
+
+    return model;
+}
+
+// Memory
+
+/**
+ * caliptra_write_u32
+ *
+ * Writes a uint32_t value to the specified address.
+ *
+ * @param[in] address Memory address to write
+ * @param[in] data Data to write at address
+ *
+ * @return 0 if successful, other if error (TBD)
+ */
+int caliptra_write_u32(uint32_t address, uint32_t data)
+{
+    struct caliptra_model *m = hwmod_get_or_init();
+
+    caliptra_model_step(m);
```

Comment:
are those steps necessary ?

Reply:
The steps down in the read/write calls have to account for all previous scenarios where a step was taking place. I can see if we're fine with only the call after the read/write to the model occurs.

Reply:
Since the test is specific for the HW Model, I believe that the caliptra_model_step() can be exposed into the test itself...